Simplified/LibraryFinder/UI/NYPLLibraryFinderViewController.swift

```diff
@@ -0,0 +1,257 @@
+                            withReuseIdentifier: searchBarReuseIdentifier)
+  }
+  
+  private func setupCollectionViewUI() {
+    collectionView.backgroundColor = NYPLConfiguration.backgroundColor()
+    self.title = NSLocalizedString("Find a Library", comment: "Title for Library Finder")
+    
+    let contentInset = NYPLLibraryFinderConfiguration.collectionViewContentInset
+    collectionView.contentInset = UIEdgeInsets(top: contentInset, left: contentInset, bottom: contentInset, right: contentInset)
+    
+    if let layout = self.collectionViewLayout as? UICollectionViewFlowLayout {
+      let insets = contentInset * 2
+      layout.estimatedItemSize = CGSize(width: collectionView.bounds.width - insets, height: 100)
+    }
+    
+    // TODO: iOS-34 Only show background label when location service not allowed
+    collectionView.backgroundView = backgroundLabel
+  }
+  
+  private func setupActivityIndicator() {
+    if #available(iOS 13.0, *) {
+      activityIndicator.color = .label
+    } else {
+      activityIndicator.style = .gray
+    }
+    
+    view.addSubview(activityIndicator)
+    activityIndicator.autoCenterInSuperview()
+    view.bringSubviewToFront(activityIndicator)
+    activityIndicator.isHidden = true
+  }
+  
+  let activityIndicator = UIActivityIndicatorView()
+  
+  let searchBarReuseIdentifier = "NYPLLibraryFinderSearchBarReuseIdentifier"
+  
+  lazy var searchBar : UISearchBar = {
+    let searchBar = UISearchBar()
+    searchBar.placeholder = NSLocalizedString("Location, library name or zip code", comment: "Placeholder text for search bar")
+    searchBar.searchBarStyle = .minimal
+    searchBar.delegate = self
+    return searchBar
+  }()
+  
+  lazy var backgroundLabel: UILabel = {
+    let label = UILabel()
+    label.numberOfLines = 0
+    label.text = NSLocalizedString("You can search for your library by name, branch location, or your own location", comment: "Tips for searching in background")
+    label.textAlignment = .center
+    label.textColor = .lightGray
+    return label
+  }()
+}
+
+// MARK: - NYPLLibraryFinderDisplaying
+
+extension NYPLLibraryFinderViewController: NYPLLibraryFinderDisplaying {
+  
+  func showMyLibrarySection() {
+    isMyLibraryHidden = false
+    UIView.animate(withDuration: 0.3, delay: 0, options: .curveEaseIn, animations: {
+      self.collectionView.reloadSections([NYPLLibraryFinderSection.myLibrary.rawValue])
+    }) { (_) in
+      // Since the library cells are dynamic height due to it's content,
+      // The starting position of "Add New Library" section could be off after the first section expanded.
+      // That's why we need to invalidate the layout after expanding the first section
+      self.collectionView.collectionViewLayout.invalidateLayout()
+    }
+  }
+  
+  func hideMyLibrarySection() {
+    isMyLibraryHidden = true
+    UIView.animate(withDuration: 0.3, animations: {
```

Comment:
can we move this number literal to a constant, maybe in the configuration class?